app/(auth)/actions.ts

```diff
@@ -0,0 +1,87 @@
+"use server";
+
+import { revalidatePath } from "next/cache";
+import { redirect } from "next/navigation";
+
+import { createClient } from "@/utils/supabase/server";
+import { SignUpWithPasswordCredentials } from "@supabase/supabase-js";
+export async function signin(formData: FormData) {
+  const supabase = createClient();
+
+  // type-casting here for convenience
+  // in practice, you should validate your inputs
+  const data = {
+    email: formData.get("email") as string,
+    password: formData.get("password") as string,
+  };
+
+  const { error } = await supabase.auth.signInWithPassword(data);
+
+  if (error) {
+    return {error: error.message };
+  }
+
+  revalidatePath("/", "layout");
+  redirect("/");
+}
+
+export async function signup(formData: FormData) {
+  const supabase = createClient();
```

Comment:
I think this supabase `createClient` function can be done once and reused as a singleton.

Reply:
Nice. Done here https://github.com/trypear/pear-landing-page/pull/20/commits/98634629462356104c0069de69f8554bfb7d9e08